src/main/js/Upload/FileUploadButton.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+import React from "react";
+import type { Branch } from "@scm-manager/ui-types";
+import { translate } from "react-i18next";
+import injectSheet from "react-jss";
+import classNames from "classnames";
+import { Link } from "react-router-dom";
+
+const styles = {
+  button: {
+    width: "50px",
+    color: "#33b2e8",
+    "&:hover": {
+      color: "#4a4a4a"
+    }
+  }
+};
+
+type Props = {
+  baseUrl: string,
+  path?: string,
+  branch?: Branch,
+  isBranchUrl: boolean,
+  // context props
+  classes: any,
+  t: string => string
+};
+
+class FileUploadButton extends React.Component<Props> {
+  createUploadUrl = () => {
+    const { baseUrl, path, branch } = this.props;
+    let uploadUrl = baseUrl.replace("sources", "upload/") + (path ? path : "");
+
+    if (branch) {
+      uploadUrl += "?branch=" + branch.name.replace("/", "%2F");
```

Comment:
Shouldn't we use `encodeURIComponent` here?

Reply:
done